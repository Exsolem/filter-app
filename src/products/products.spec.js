import React from 'react'
import {shallow} from 'enzyme'

import Products from './products'


describe('Products', () => {
    const mockGetCurrentCategory = jest.fn();
    const props = {
        getCurrentCategory: mockGetCurrentCategory,
        search:'',
        filteredProducts: [],
        searchedProducts:[],
        products: [],
        history: {
            location:{
                pathname:'/'
            }
        }
    };
    describe('Products initial', () => {
        const products = shallow(<Products {...props}/>);
        it('renders properly', () => {
            expect(products).toMatchSnapshot()
        });
        it('getCurrentCategory func was called',() => {
            expect(mockGetCurrentCategory).toHaveBeenCalled();
        })
        it('Product not rendered', () => {
            expect(products.find('Product')).toHaveLength(0);
        })
    });
    describe('Products data received',() => {
        describe('Products received', () => {
            const nextProps = {
                ...props,
                products: [1,2,3]
            };
            const products = shallow(<Products {...nextProps}/>);

            it('Product rendered', () => {
                expect(products.find('Product')).toHaveLength(3);
            })
        });
        describe('Filtered Products received', () => {
            const nextProps = {
                ...props,
                filteredProducts: [1,2,3]
            };
            const products = shallow(<Products {...nextProps}/>);

            it('Product rendered', () => {
                expect(products.find('Product')).toHaveLength(3);
            })
        });
        describe('Searched Products received', () => {
            const nextProps = {
                ...props,
                filteredProducts: [1,2,3]
            };
            const products = shallow(<Products {...nextProps}/>);

            it('Product rendered', () => {
                expect(products.find('Product')).toHaveLength(3);
            })
        })

    })
});