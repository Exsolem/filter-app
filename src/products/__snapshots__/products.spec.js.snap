// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products Products initial renders properly 1`] = `ShallowWrapper {}`;
